### PR TITLE
Performance measurements: `bootCmp` resolves after fully initialised

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Refactor `bootCmp` to only resolve once all the consent management platform (CMP) work is complete:
- CMP initialised
- privacy link injected
- Ophan events sent

## Why?

Start of a piece of work to get meaningful timings from startup scripts.
Part of @guardian/open-journalism’s reporting on performance: #8529

### CODE REVIEW

File: dotcom-rendering/src/lib/DOMContentLoaded.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ */
+export const DOMContentLoaded = new Promise<void>((resolve) => {
+	if (document.readyState === 'loading') {
+		document.addEventListener('DOMContentLoaded', () => resolve(), {
+			once: true,
+		});
+	} else {
+		resolve();
+	}
+});
```
Comment (mxdvl) [inline dotcom-rendering/src/lib/DOMContentLoaded.ts:14]: This is a refactor from @domlander’s work in #8343

Comment (domlander) [reply dotcom-rendering/src/lib/DOMContentLoaded.ts:14]: Yep, it looks good to me.

The [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState) claim that the document ready state will be 'interactive' _before_ the `DOMContentLoaded` event fires, so I think we're all good.

**Comment by sndrs** [comment]
i think this is ok but i paired on it, so don't want to approve it :) but @guardian/transparency-consent does it look ok to you?

**Review by sndrs — APPROVED** [review]

**Review by akinsola-guardian — APPROVED** [review]
LGTM. Tested locally and seemed to work as expected.

**Pull request merged**